src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -0,0 +1,115 @@
+//     { scheme: 'file', language: PYTHON_LANGUAGE },
+//     { scheme: 'untitled', language: PYTHON_LANGUAGE }
+// ]
+export const PYTHON_ALLFILES = [
+    { language: PYTHON_LANGUAGE }
+]
+
+export const getSamPythonProjectDirPath = async (): Promise<string> => {
+    const activeFilePath = vscode.window.activeTextEditor!.document.uri.fsPath
+    if (!activeFilePath) {
+      throw new Error('"vscode.window.activeTextEditor" not defined')
+    }
+
+    return path.dirname(activeFilePath)
+}
+
+export const initialize = ({
+    configuration,
+    toolkitOutputChannel,
+    processInvoker = new DefaultSamCliProcessInvoker(),
+    taskInvoker = new DefaultSamCliTaskInvoker()
+}: CodeLensProviderParams): void => {
+    vscode.commands.registerCommand(
+        getInvokeCmdKey('python'),
+        async (args: LambdaLocalInvokeArguments) => {
+            const samProjectCodeRoot = await getSamPythonProjectDirPath()
+            logger.debug(`Project root: '${samProjectCodeRoot}'`)
+            let debugPort: number | undefined
+
+            if (args.debug) {
+                debugPort = 5858
+            }
+
+            const debugConfig: NodeDebugConfiguration = {
+                type: PYTHON_LANGUAGE,
+                request: 'attach',
+                name: 'SamLocalDebug',
+                preLaunchTask: undefined,
+                address: 'localhost',
+                port: debugPort!,
+                localRoot: samProjectCodeRoot,
+                remoteRoot: '/var/task',
+                protocol: 'inspector',
+                skipFiles: []
+            }
+
+            const localLambdaRunner: LocalLambdaRunner = new LocalLambdaRunner(
+                configuration,
+                args,
+                debugPort,
+                'python3.7', // TODO: Remove hard coded value
+                toolkitOutputChannel,
+                processInvoker,
+                taskInvoker,
+                debugConfig,
+                samProjectCodeRoot
+
+            )
+
+            await localLambdaRunner.run()
+        }
+    )
+}
+
+export const makePythonCodeLensProvider = (): vscode.CodeLensProvider => {
+    return { // CodeLensProvider
+        provideCodeLenses: async (
+            document: vscode.TextDocument,
+            token: vscode.CancellationToken
+        ): Promise<vscode.CodeLens[]> => {
+            const handlers: LambdaHandlerCandidate[] = await getLambdaHandlerCandidates({uri: document.uri})
+
+            return makeCodeLenses({
+                                      document,
+                                      handlers,
+                                      token,
+                                      lang: 'python'
+                                  })
+        },
+        resolveCodeLens: (
+            codeLens: vscode.CodeLens,
+            token: vscode.CancellationToken
+        ): vscode.ProviderResult<vscode.CodeLens> => {
+            throw new Error('not implemented')
```

Comment:
Please change this to something like `resolveCodeLens not implemented` - it will help us distinguish where the issue is if it ever arises.

Reply:
✔